protographic/tests/sdl-to-proto/11-lists.test.ts

```diff
@@ -0,0 +1,1222 @@
+      }
+
+      // Wrapper message for a list of Priority.
+      message ListOfListOfListOfListOfPriority {
+        message List {
+          repeated ListOfListOfListOfPriority items = 1;
+        }
+        List list = 1;
+      }
+
+      // Wrapper message for a list of Priority.
+      message ListOfListOfListOfPriority {
+        message List {
+          repeated ListOfListOfPriority items = 1;
+        }
+        List list = 1;
+      }
+
+      // Wrapper message for a list of Priority.
+      message ListOfListOfPriority {
+        message List {
+          repeated ListOfPriority items = 1;
+        }
+        List list = 1;
+      }
+
+      // Wrapper message for a list of String.
+      message ListOfListOfString {
+        message List {
+          repeated ListOfString items = 1;
+        }
+        List list = 1;
+      }
+
+      // Wrapper message for a list of Node.
+      message ListOfNode {
+        repeated Node items = 1;
+      }
+
+      // Wrapper message for a list of Priority.
+      message ListOfPriority {
+        repeated Priority items = 1;
+      }
+
+      // Wrapper message for a list of String.
+      message ListOfString {
+        repeated string items = 1;
+      }
+
+      // Request message for getComplex operation.
+      message QueryGetComplexRequest {
+        FilterInput filter = 1;
+      }
+      // Response message for getComplex operation.
+      message QueryGetComplexResponse {
+        ComplexType get_complex = 1;
+      }
+
+      message FilterInput {
+        ListOfPriority priorities = 1;
+        ListOfID user_ids = 2;
+      }
+
+      message ComplexType {
+        repeated string strings = 1;
+        ListOfString optional_strings = 2;
+        repeated Priority priorities = 3;
+        ListOfPriority optional_priorities = 4;
+        repeated Node nodes = 5;
+        ListOfNode optional_nodes = 6;
+        repeated Item items = 7;
+        ListOfItem optional_items = 8;
+        ListOfListOfString nested_strings = 9;
+        ListOfListOfString nested_optional_strings = 10;
+        ListOfListOfPriority nested_priorities = 11;
+        ListOfListOfPriority nested_optional_priorities = 12;
+        ListOfListOfListOfItem deep_nested_items = 13;
+        ListOfListOfListOfListOfPriority mixed_nested = 14;
+      }
+
+      enum Priority {
+        PRIORITY_UNSPECIFIED = 0;
+        PRIORITY_LOW = 1;
+        PRIORITY_MEDIUM = 2;
+        PRIORITY_HIGH = 3;
+      }
+
+      message Node {
+        oneof instance {
+        User user = 1;
+        Task task = 2;
+        }
+      }
+
+      message User {
+        string id = 1;
+        string name = 2;
+      }
+
+      message Task {
+        string id = 1;
+        string title = 2;
+      }
+
+      message Item {
+        oneof value {
+        User user = 1;
+        Task task = 2;
+        }
+      }"
+    `);
+  });
+
+  it('should correctly generate protobuf for edge cases with empty lists and optional nesting', () => {
```

Comment:
Do we have tests that cover recursion and input types?

Reply:
done